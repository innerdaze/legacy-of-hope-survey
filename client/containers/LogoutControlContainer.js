import { connect } from 'react-redux'
import LogoutControl from '../components/LogoutControl'
//import { logoutCashier } from '../actions/CashierActions'
import { cashierOperations } from '../features/cashier'
export default connect(null, dispatch => ({
  handleLogoutPress: () => {
    dispatch(cashierOperations.logoutCashier())
  }
}))(LogoutControl)
