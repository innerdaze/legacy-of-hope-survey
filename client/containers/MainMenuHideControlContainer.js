import { connect } from 'react-redux'
import { uiOperations } from '../features/ui'
import MainMenuHideControl from '../components/MainMenuHideControl'

const mapDispatchToProps = dispatch => {
  return {
    hideMenu: () => dispatch(uiOperations.uiHideMenu())
  }
}

export default connect(null, mapDispatchToProps)(MainMenuHideControl)
