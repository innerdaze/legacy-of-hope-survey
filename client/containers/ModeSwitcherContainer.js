import { connect } from 'react-redux'
import { orderOperations, orderSelectors } from '../features/order'
import { uiOperations } from '../features/ui'
import ModeSwitcher from '../components/ModeSwitcher'

const mapDispatchToProps = dispatch => {
  return {
    onSwitch: mode => {
      dispatch(orderOperations.changeOperationMode(mode))
      dispatch(uiOperations.uiHideMenu())
    }
  }
}

const ModeSwitcherContainer = connect(
  state => ({
    activeMode: orderSelectors.modeSelector(state)
  }),
  mapDispatchToProps
)(ModeSwitcher)

export default ModeSwitcherContainer
