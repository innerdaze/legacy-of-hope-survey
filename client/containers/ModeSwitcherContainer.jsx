import { connect } from 'react-redux'
import { changeOperationMode } from '../actions/OrderActions'
import ModeSwitcher from '../components/ModeSwitcher.jsx'

const mapStateToProps = state => {
  return {}
}

const mapDispatchToProps = dispatch => {
  return {
    onSwitch: mode => {
      dispatch(changeOperationMode(mode))
    }
  }
}

const ModeSwitcherContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ModeSwitcher)

export default ModeSwitcherContainer
