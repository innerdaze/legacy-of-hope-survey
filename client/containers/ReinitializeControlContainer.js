import { connect } from 'react-redux'
import ReinitializeControl from '../components/ReinitializeControl'
import { appOperations } from '../features/app'

export default connect(null, dispatch => ({
  handleReinitializePress: () => {
    dispatch(appOperations.reset())
  }
}))(ReinitializeControl)
