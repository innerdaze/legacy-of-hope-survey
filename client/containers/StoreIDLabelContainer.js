import { connect } from 'react-redux'
import StoreIDLabel from '../components/StoreIDLabel'
import {appSelectors} from '../ducks/app'
const mapStateToProps = state => {
  return {
    storeID: appSelectors.storeID(state)
  }
}

export default connect(
  mapStateToProps
)(StoreIDLabel)
