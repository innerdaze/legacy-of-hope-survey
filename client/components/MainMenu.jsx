import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Sidebar from 'grommet/components/Sidebar'
import Header from 'grommet/components/Header'
import Box from 'grommet/components/Box'
import Title from 'grommet/components/Title'
import MainMenuHideControlContainer from '../containers/MainMenuHideControlContainer'
import ModeSwitcherContainer from '../containers/ModeSwitcherContainer'
import Modes from '../constants/OperationModes'

const MainMenu = () => (
  <Sidebar colorIndex='neutral-1' full={true}>
    <Header pad='medium'
      justify='between'>
      <Title>
        Site Assistant
      </Title>
      <MainMenuHideControlContainer/>
    </Header>
    <Box flex='grow'
      justify='start'>
      <ModeSwitcherContainer/>
    </Box>
  </Sidebar>
)

export default MainMenu
