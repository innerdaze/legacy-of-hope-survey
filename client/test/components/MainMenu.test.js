import React from 'react'
import MainMenu from '../../components/MainMenu'
import MainMenuHideControlContainer from '../../containers/MainMenuHideControlContainer'
import StoreIDLabelContainer from '../../containers/StoreIDLabelContainer'
import ModeSwitcherContainer from '../../containers/ModeSwitcherContainer'
import ReinitializeControlContainer from '../../containers/ReinitializeControlContainer'
import LogoutControlContainer from '../../containers/LogoutControlContainer'

let wrapper

describe('Testing on MainMenu...', () => {
  beforeEach(() => {
    wrapper = shallow(<MainMenu />)
  })

  describe('MainMenu layout', () => {
    test('Expect MainMenu to have parent Sidebar comp and Header, Title and 4 Box components', () => {
      expect(wrapper.find('Sidebar')).toHaveLength(1)
      expect(wrapper.find('Sidebar').find('Header')).toHaveLength(1)
      expect(wrapper.find('Sidebar').find('Box')).toHaveLength(4)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Header to have a Title and MainMenuHideControlContainer in it', () => {
      expect(
        wrapper
          .find('Sidebar')
          .find('Header')
          .find('Title')
      ).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()

      expect(
        wrapper
          .find('Sidebar')
          .find('Header')
          .contains(<MainMenuHideControlContainer />)
      ).toEqual(true)
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Box component to have StoreIDLabelContainer, ModeSwitcherContainer, ReinitializeControlContainer and LogoutControlContainer', () => {
      expect(
        wrapper
          .find('Sidebar')
          .find('Box')
          .contains(<StoreIDLabelContainer />)
      ).toEqual(true)
      expect(wrapper).toMatchSnapshot()

      expect(
        wrapper
          .find('Sidebar')
          .find('Box')
          .contains(<ModeSwitcherContainer />)
      ).toEqual(true)
      expect(wrapper).toMatchSnapshot()

      expect(
        wrapper
          .find('Sidebar')
          .find('Box')
          .contains(<ReinitializeControlContainer />)
      ).toEqual(true)
      expect(wrapper).toMatchSnapshot()

      expect(
        wrapper
          .find('Sidebar')
          .find('Box')
          .contains(<LogoutControlContainer />)
      ).toEqual(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
