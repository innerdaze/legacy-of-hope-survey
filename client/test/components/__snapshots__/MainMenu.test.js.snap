// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on MainMenu... MainMenu layout Expect Box component to have StoreIDLabelContainer, ModeSwitcherContainer, ReinitializeControlContainer and LogoutControlContainer 1`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect Box component to have StoreIDLabelContainer, ModeSwitcherContainer, ReinitializeControlContainer and LogoutControlContainer 2`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect Box component to have StoreIDLabelContainer, ModeSwitcherContainer, ReinitializeControlContainer and LogoutControlContainer 3`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect Box component to have StoreIDLabelContainer, ModeSwitcherContainer, ReinitializeControlContainer and LogoutControlContainer 4`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect Header to have a Title and MainMenuHideControlContainer in it 1`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect Header to have a Title and MainMenuHideControlContainer in it 2`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;

exports[`Testing on MainMenu... MainMenu layout Expect MainMenu to have parent Sidebar comp and Header, Title and 4 Box components 1`] = `
<Sidebar
  colorIndex="brand"
  direction="column"
  full={true}
>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Site Assistant
    </Title>
    <Connect(MainMenuHideControl) />
  </Header>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Connect(StoreIDLabel) />
  </Box>
  <Box
    announce={false}
    direction="column"
    flex="grow"
    focusable={true}
    justify="start"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ModeSwitcher) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(ReinitializeControl) />
  </Box>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Connect(LogoutControl) />
  </Box>
</Sidebar>
`;
